e2e/machine_migration_capi_authoritative_test.go

```diff
@@ -161,11 +161,84 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 				})
 				It("should verify the AWS machine is deleted", func() {
 					verifyResourceRemoved(&awsv1.AWSMachine{
-						TypeMeta: metav1.TypeMeta{Kind: "AWSMachine", APIVersion: awsv1.GroupVersion.String()},
+						TypeMeta:   metav1.TypeMeta{Kind: "AWSMachine", APIVersion: awsv1.GroupVersion.String()},
 						ObjectMeta: metav1.ObjectMeta{Name: mapiMachineAuthCAPINameDeletion, Namespace: capiframework.CAPINamespace},
 					})
 				})
 			})
 		})
 	})
+
+	var _ = Describe("Machine Migration Round Trip Tests", Ordered, func() {
+		var capiMapiCapiRoundTripName = "machine-capi-mapi-capi-roundtrip"
+		var newMapiMachine *mapiv1beta1.Machine
+		var newCapiMachine *clusterv1.Machine
+
+		Context("CAPI -> MAPI -> CAPI round trip", func() {
+			BeforeAll(func() {
+				DeferCleanup(func() {
+					By("Cleaning up machine resources")
+					cleanupMachineResources(
+						ctx,
+						cl,
+						[]*clusterv1.Machine{newCapiMachine},
+						[]*mapiv1beta1.Machine{newMapiMachine},
+					)
+				})
+			})
+
+			It("should create MAPI Machine with spec.authoritativeAPI: ClusterAPI", func() {
```

Comment:
One problem here IMO, which is that we are abusing the `It()` construct. We should be describing a behaviour, we're conflating `It()` with `By()`. This makes reading the test a little confusing (maybe just for me?) 

We could move the creation of the MAPI machine to our `BeforeAll()`.

So something like this (descriptions can be improved, I'm sure)

```go
It("creates a CAPI mirror machine", func() {
newCapiMachine = capiframework.GetMachine(cl, capiMapiCapiRoundTripName, capiframework.CAPINamespace)
verifyMachineRunning(cl, newCapiMachine)
})